File: src/providers/fireworks.ts
import { makeOpenAIAdapter, type OpenAIAdapterConfig } from "./shared";

export const fireworksConfig = {
  id: "fireworks",
  displayName: "Fireworks AI",
  baseUrl: "https://api.fireworks.ai/inference/v1",
  credentialKind: "api_key",
  credentialUrl: "https://fireworks.ai/account/api-keys",
} as const satisfies OpenAIAdapterConfig;

export const FireworksAdapter = makeOpenAIAdapter(fireworksConfig);

File: router/internal/providers/builtin/fireworks.go
package builtin

func Fireworks() ProviderDefinition {
	return ProviderDefinition{
		ID: "fireworks", DisplayName: "Fireworks AI", Protocol: ProtocolOpenAI,
		Adapter: AdapterOpenAI, CredentialKind: CredentialAPIKey,
		BaseURL: "https://api.fireworks.ai/inference/v1", Surfaces: []Surface{SurfaceOpenAIChat},
		Models: []ProviderModel{Model("kimi-k2.7-code", "kimi-k2.7-code", nil)},
	}
}
